src/main/resources/db/changesets/20231129_snapshotrequests.yaml

```diff
@@ -0,0 +1,60 @@
+databaseChangeLog:
+  - changeSet:
+      id: snapshot_requests
+      author: rjohanek
+      changes:
+        - createTable:
+            tableName: snapshot_requests
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: dataset_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_dataset_snapshot_requests
+                    references: dataset(id)
+                    deleteCascade: true
+              - column:
+                  name: request_name
```

Comment:
I think the intent is that it's not the _request_ name, but the created snapshot name. So `snapshot_name` might be better here. If it were the request's name, then using `name` would be sufficient since this is the `request` table, and its attributes refer to itself by default.